### PR TITLE
Layers num phase function

### PR DESCRIPTION
**Pull request by jtgasparik**
Edited and tested num_phase_instances function to include layers

### CODE REVIEW

File: src/aero_reps/aero_rep_single_particle_layers.F90 [hunk 1]
```diff
@@ -632,25 +632,26 @@ end function spec_name
   !> Get the number of instances of a specified aerosol phase. In the single
   !! particle representation, if an aerosol phase is present, is exists once
   !! in each computational particle.
```
Comment (mattldawson) [inline src/aero_reps/aero_rep_single_particle_layers.F90]: This description looks like it needs to be updated for the layered particles

File: test/unit_aero_rep_data/test_aero_rep_single_particle_layers.F90 [hunk 2]
```diff
@@ -147,7 +147,25 @@ subroutine test_ordered_layer_ids()
...
+!    num_phases = 2
+    ! Call the function and enter the inputs 
+!    num_phase_instances = aero_rep%num_phase_instances(phase_name_test) 
+    
+    ! check value
+!    call assert(734138496, num_phases .eq. num_phase_instances)
+
+!  end subroutine test_num_phase_instances
 
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```
Comment (mattldawson) [inline test/unit_aero_rep_data/test_aero_rep_single_particle_layers.F90]: please delete commented out code

**Review by mattldawson — CHANGES_REQUESTED** [review]
looks great! just a couple minor comments

File: src/aero_reps/aero_rep_single_particle_layers.F90 (hunk 1 above)
Comment (jtgasparik) [reply src/aero_reps/aero_rep_single_particle_layers.F90]: done! https://github.com/jtgasparik/camp_/blob/203ff906061f055a5466d151f0b7c1dbfc7aa574/src/aero_reps/aero_rep_single_particle_layers.F90#L632C2-L634C22

File: test/unit_aero_rep_data/test_aero_rep_single_particle_layers.F90 (hunk 2 above)
Comment (jtgasparik) [reply test/unit_aero_rep_data/test_aero_rep_single_particle_layers.F90]: done!

File: src/aero_reps/aero_rep_single_particle_layers.F90 [hunk 3]
```diff
@@ -668,6 +669,7 @@ function num_jac_elem(this, phase_id)
 
     integer(kind=i_kind) :: i_phase
 
+    ! TODO : does this need to be updated?
```
Comment (mattldawson) [inline src/aero_reps/aero_rep_single_particle_layers.F90]:
```suggestion
```

**Review by mattldawson — CHANGES_REQUESTED** [review]

**Review by mattldawson — APPROVED** [review]

**Review by jcurtis2 — APPROVED** [review]

**Pull request merged**